sopel/bot.py

```diff
@@ -530,6 +530,9 @@ def dispatch(self, pretrigger):
                                 match = True
                         if not match:
                             continue
+                    if (not (hasattr(func, 'echo') and func.echo is True) and
```

Comment:
You can replace this by `not getattr(func, 'echo', False)`.

Reply:
I changed the loader's `clean_callable` to always set the `echo` attribute, so I need to revisit this line anyway.

Reply:
Done in latest force-push. Now this just checks `not func.echo`, because the loader fills in that attribute (as `False`) for undecorated callables.